desc/compute/_geometry.py

```diff
@@ -466,6 +466,46 @@ def _perimeter_of_z(params, transforms, profiles, data, **kwargs):
     return data
 
 
+@register_compute_fun(
+    name="perimeter(z)",
+    label="P(\\zeta)",
+    units="m",
+    units_long="meters",
+    description="Perimeter of enclosed cross-section (enclosed constant zeta surface), "
+    "scaled by max(ρ)⁻¹, as function of zeta",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="z",
+    data=["rho", "e_theta"],
+    parameterization=[
+        "desc.geometry.surface.ZernikeRZToroidalSection",
+    ],
+    resolution_requirement="rt",  # just need max(rho) near 1
+)
+def _perimeter_of_z_ZernikeRZToroidalSection(
+    params, transforms, profiles, data, **kwargs
+):
+    # NOTE: this is perimeter of a constant zeta XS
+    # as a ZernikeRZToroidalSection is defined at constant zeta, so
+    # doing a perimeter at constant phi would make no sense.
```

Comment:
Originally, these quantities were designed to be computed at constant phi for the elongation. So although the description says constant zeta surface, the current user expectation has been that the computation be done at constant phi surface.

Frankly, these should either be renamed to `perimter(phi)` (no use case where you compute constant phi stuff on grids that are uniform in zeta not phi), or we should decide that we only care about constant zeta sections rather than constant phi cross sections.